crates/biome_cli/src/reporter/mod.rs

```diff
@@ -0,0 +1,206 @@
+pub(crate) mod terminal;
+
+use crate::cli_options::CliOptions;
+use crate::execute::Execution;
+use crate::CliDiagnostic;
+use biome_diagnostics::{Error, Severity};
+use std::io;
+use std::time::Duration;
+
+pub struct DiagnosticsPayload<'a> {
+    diagnostics: Vec<Error>,
+    verbose: bool,
+    diagnostic_level: Severity,
+    execution: &'a Execution,
+}
+
+/// A type that holds the result of the traversal
+#[derive(Debug, Default)]
+pub struct TraversalSummary {
+    changed: usize,
```

Comment:
Should we maybe track `processed` instead of `unchanged`? I'm thinking `processed = changed + unchanged`, so it would change the method of tracking slightly, but I feel `changed`/`unchanged` feels a little bit like a mismatch for use cases that aren't intended to change files in the first place, such as `biome ci` or the upcoming `biome search`.

Reply:
We can, although not in this PR because It seems out of scope